File: lab3/part2/part2b.R
# read in data from CSV
known_spectra <-
  read.csv("spectra/known_spectra.csv", row.names = "wavelength_A")
unknown_spectra <-
  read.csv("spectra/unknown_spectra.csv", row.names = "wavelength_A")

# instantiate data frame with known spectral types as row names
residual_sumsquares <-
  data.frame(spectral_type = colnames(known_spectra))

# populate data frame of residual sum of squares
for (current_unknown_spectra in colnames(unknown_spectra))
  residual_sumsquares[[current_unknown_spectra]] <-
  as.vector(colSums((unknown_spectra[[current_unknown_spectra]] - known_spectra) ^
                      2))

# write to CSV file
write.csv(residual_sumsquares,
          "part2/residual_sumsquares.csv",
          row.names = FALSE)

# print final results
for (current_unknown_spectra in colnames(unknown_spectra))
  print(paste(current_unknown_spectra, residual_sumsquares$spectral_type[residual_sumsquares[[current_unknown_spectra]] == min(residual_sumsquares[[current_unknown_spectra]])], min(residual_sumsquares[[current_unknown_spectra]])))

File: lab3/part2/part2b.m
% read in data from CSV
known_spectra = readtable('known_spectra.csv', 'ReadVariableNames', true, 'ReadRowNames', true);
unknown_spectra = readtable('unknown_spectra.csv', 'ReadVariableNames', true, 'ReadRowNames', true);

residual_sumsquares = array2table(zeros(12, 5), 'VariableNames', unknown_spectra.Properties.VariableNames, 'RowNames', known_spectra.Properties.VariableNames);

for current_unknown_spectra = unknown_spectra.Properties.VariableNames
    for current_known_spectra = known_spectra.Properties.VariableNames
        residual_sumsquares{:, current_unknown_spectra} = sum((table2array(unknown_spectra(:, current_unknown_spectra)) - table2array(known_spectra)).^2)';
    end
end

